port_drayage_plugin/src/port_drayage_worker.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2018-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "port_drayage_plugin/port_drayage_worker.h"
+#include "ros/ros.h"
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+namespace port_drayage_plugin
+{
+    bool PortDrayageWorker::check_for_stop(const cav_msgs::ManeuverPlanConstPtr& plan, const geometry_msgs::TwistStampedConstPtr& speed) {
+        if (plan == nullptr || speed == nullptr) {
+            ROS_WARN("Checking for stop when PortDrayagePlugin not properly initialized. Speed or plan is null");
+            return false;
+        }
+
+        if (plan->maneuvers.size() > 0) {
+            if (plan->maneuvers[0].type == cav_msgs::Maneuver::STOP_AND_WAIT) {
+                cav_msgs::Maneuver stop_maneuver = _cur_plan->maneuvers[0];
+                if (stop_maneuver.stop_and_wait_maneuver.parameters.planning_strategic_plugin == PORT_DRAYAGE_PLUGIN_ID) {
+                    double longitudinal_speed = speed->twist.linear.x; // TODO: Verify that X is longitudinal speed of vehicle
```

Comment:
yes, twist.linear.x is the longitudinal speed of the vehicle. Please make sure the "speed" is twist type